codes/zig/include/TreeNode.zig

```diff
@@ -28,32 +28,26 @@ pub fn arrToTree(comptime T: type, mem_allocator: std.mem.Allocator, list: []T)
     if (list.len == 0) return null;
     var root = try mem_allocator.create(TreeNode(T));
     root.init(list[0]);
-
-    const TailQueue = std.TailQueue(?*TreeNode(T));
-    const TailQueueNode = std.TailQueue(?*TreeNode(T)).Node;
-    var que = TailQueue{};
-    var node_root = TailQueueNode{ .data = root };
-    que.append(&node_root); 
+    var que = std.ArrayList(*TreeNode(T)).init(std.heap.page_allocator);
+    try que.append(root); 
     var index: usize = 0;
-    while (que.len > 0) {
-        var node = que.popFirst();
```

Comment:
I'm curious why we cannot use the `TailQueue()` here. We'd better use queue rather than List.

Reply:
> I'm curious why we cannot use the `TailQueue()` here. We'd better use queue rather than List.

Now, the codes  of this PR were updated. 
Now， it already supports TailQueue, and also support ArrayList to work as queue. 
The output of  "zig/chapter_heap/heap.zig" using TailQueue is the same as using  ArrayList to work as queue.
So the readers can select one of the options trough  "queue_flag" parameter in "printHeap" function in "zig/include/PrintUtil.zig" file.